### PR TITLE
Improve REMReM introduction description

### PR DESCRIPTION
**Pull request by m-linner-ericsson**
### Applicable Issues
None

### Description of the Change
Add an introduction description of what REMReM does. The introduction also spells out what REMReM stands for.

Other things fixed:
* Fix spelling of REMReM
* Remove reference to REMReM shared (not used)
* Removed non needed copyright headers

### Alternate Designs
Couldn't find any other better place to put this information.

### Benefits
New readers can understand what REMReM does. The current description did not give that introduction.

### Possible Drawbacks
None that I can think of

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Mattias Linnér mattias.linner@ericsson.com

### CODE REVIEW

**Review by e-backmark-ericsson — APPROVED** [review]

File: overview.md [hunk 1]
```diff
@@ -29,11 +19,10 @@ Architecture of remrem is based on Microservice and API design principles utiliz
...
-- Shared (Library): Interface information used in injecting message libraries. Utilized by Semantics and Eifel3Messaging innersource projects [Shared Github Repo](https://github.com/eiffel-community/eiffel-remrem-shared)
 
-# Building and releasing
-For every RemRem component it is needed to create a new release tag in the component's repository on github. Once the tag is created the build for the new release tag will be started when you visit the component's page on jitpack.io (i.e. https://jitpack.io/#eiffel-community/eiffel-remrem-generate) and published on component's page on jitpack.io or when another application is build and it is dependent of the new release tag from jitpack.io.
+## Building and releasing
+For every REMRem component it is needed to create a new release tag in the component's repository on GitHub. Once the tag is created the build for the new release tag will be started when you visit the component's page on jitpack.io (i.e. https://jitpack.io/#eiffel-community/eiffel-remrem-generate) and published on component's page on jitpack.io or when another application is build and it is dependent of the new release tag from jitpack.io.
```
Comment (SantoshNC68) [inline overview.md:28]: REMReM

**Review by SantoshNC68 — CHANGES_REQUESTED** [review]
Also one more thing, in the architecture diagram it is pointing to shared library, has that been modified ? If not the picture has to be modified to reflect the current behavior.

File: overview.md (hunk 1 above)
Comment (m-linner-ericsson) [reply overview.md]: Thanks! Fixed in 9f0e2d9

**Comment by m-linner-ericsson** [comment]
> Also one more thing, in the architecture diagram it is pointing to shared library, has that been modified ? If not the picture has to be modified to reflect the current behavior.

Thanks! Fixed in 9f0e2d9

**Review by SantoshNC68 — COMMENTED** [review]
Just one more little thing on the diagram, there should be a dotted line for other protocol as well as eiffel-remrem-semantics connecting to eiffel-remrem-protocol-interface.

**Review by SantoshNC68 — COMMENTED** [review]
Any protocol trying to comply with eiffel should try to create the services based on the eiffel-remrem-protocol-interface.

**Review by k-hallen-ericsson — APPROVED** [review]

**Comment by m-linner-ericsson** [comment]
@SantoshNC68 In https://github.com/eiffel-community/eiffel-remrem-publish/blob/master/pom.xml#L13 it looks like publish uses both semantics and shared. Same with https://github.com/eiffel-community/eiffel-remrem-generate/blob/master/pom.xml#L14. Is the picture then correct?

**Comment by SantoshNC68** [comment]
> @SantoshNC68 In https://github.com/eiffel-community/eiffel-remrem-publish/blob/master/pom.xml#L13 it looks like publish uses both semantics and shared. Same with https://github.com/eiffel-community/eiffel-remrem-generate/blob/master/pom.xml#L14. Is the picture then correct?


The eiffel-remrem-shared earlier used to have the code for protocol interfaces that needed to be extended in eiffel-remrem-semantics, but that has changed and now any protocol that should define its own definition of generate, version, etc needs to extend eiffel-remrem-protocol-interface.

There are left over code like VersionService which is used to identify the semantics/any other versions using endpoints using the VersionService class present in the shared library. But the picture will still remain the same as I suggested that the semantics/other protocols should have a dotted connection to eiffel-remrem-protocol-interface.

**Comment by m-linner-ericsson** [comment]
@SantoshNC68 I have now in 5e05d85 updated the image. I guess that the POMs for generate and publish need updates?

**Review by SantoshNC68 — APPROVED** [review]

**Comment by e-backmark-ericsson** [comment]
@SantoshNC68 , will you write issues on generate and publish to update their POMs accordingly? We shouldn't remove the shared repo until all references to it are gone.

**Pull request merged**